File: app/src/main/java/com/icerrate/popularmovies/provider/cloud/ServiceRequest.java
package com.icerrate.popularmovies.provider.cloud;

import com.icerrate.popularmovies.view.common.BaseCallback;

/**
 * @author Ivan Cerrate
 */
public abstract class ServiceRequest<T> {

    abstract void enqueue(BaseCallback<T> doctavioCallback);

    public abstract void cancel();
}

File: app/src/main/java/com/icerrate/popularmovies/provider/cloud/ServiceRequest.kt
package com.icerrate.popularmovies.provider.cloud

import com.icerrate.popularmovies.view.common.BaseCallback

/**
 * @author Ivan Cerrate.
 */
abstract class ServiceRequest<T> {
    abstract fun enqueue(callback: BaseCallback<T>)
    abstract fun cancel()
}